### ISSUE TITLE
"docker exec $container ps" doesn't work without -it

### ISSUE THREAD

**Issue by dlespiau** [01org/cc-oci-runtime#535]
I start a container with:

```
docker run busybox sh -c "while :; do echo .; sleep 5; done"
```

Then, this works just fine:

```
docker exec -it $container ps
```

But the following doesn't:

```
$ docker exec loving_hugle ps
rpc error: code = 2 desc = containerd: container not started
```

**Comment by sboeuf** [comment]
@dlespiau , @jcvenegas have you been able to reproduce it ? It is working fine on my laptop... No need to add -it, I get the result of the ps without any error.

**Comment by jcvenegas** [comment]
@dlespiau  @sboeuf , same here I have tried this case but seems that works well.

But tryting to see if was a radom bug I did
```
while sudo docker exec $(sudo docker ps -ql) ps;do echo ok; done
```
and after a while and exec workload  hang as the container

**Comment by sboeuf** [comment]
@jcvenegas if it hangs after a while, this means this could be something fixed by @sameo PR #538. But you confirm that it does work for you.
I mean if you look at @dlespiau issue description, it is clear that he could not even make this work once.

**Comment by jcvenegas** [comment]
@sboeuf yes it is working, I was not able to reproduce @dlespiau  issue

**Comment by sboeuf** [comment]
Let's close this one then :)

**Comment by jcvenegas** [comment]
sure, lets re-open it if someone else see this behavior.

**Comment by jcvenegas** [comment]
reopen the issue was seen in #538

**Comment by devimc** [comment]
seems like this issue was fixed in #538
